### PR TITLE
Docs: Refactored Gestalt Docs Sidebar Menu

### PR DESCRIPTION
**Pull request by AlbertCarreras**
- Refactored sidebar menu implementing:

1. grouping by component type
2. menu sections (such as `Getting Started` and `Components`)
3. sidebar menu responsiveness

## TODO

- [-] Accessibility checkup
- [x] Update documentation
- [-] Add/update Tests
- [-] Pinterest Designer (for design changes)
- [x] [Format PR title](https://github.com/pinterest/gestalt/#releasing): `ComponentName: Description`.

### CODE REVIEW

File: docs/src/1-Installation.doc.js [hunk 1]
```diff
@@ -0,0 +1,259 @@
+// @flow strict
```
Comment (christianvuerings) [inline docs/src/1-Installation.doc.js:1]: @AlbertCarreras let's avoid the numbering of the files.

File: docs/src/1-Installation.doc.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+~~~
+  "
+      />
+    </Box>
+  </Stack>
+);
+
+export default cards;
+
+const navRoute = { section: 'getting-started', group: 'none' };
```
Comment (christianvuerings) [inline docs/src/1-Installation.doc.js:258]:
```
export {
  subSection: null,
  section: 'getting-started',
}
```

File: docs/src/2-FAQs.doc.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// @flow strict
+import * as React from 'react';
+import { Heading, Text } from 'gestalt';
+
+const cards = [];
+const card = c => cards.push(c);
+
+card(<Heading>FAQs</Heading>);
+card(<Text>COMING SOON! </Text>);
```
Comment (christianvuerings) [inline docs/src/2-FAQs.doc.js:9]: @AlbertCarreras let's remove the coming soon pages?

File: docs/src/components/App.js [hunk 4]
```diff
@@ -1,43 +1,52 @@
...
+      }}
+    >
+      <Box minHeight="100vh">
+        <Header />
+        <Box mdDisplay="flex" direction="row">
+          <Column span={12} mdSpan={2}>
+            <Navigation />
+          </Column>
           <Divider />
+          <Column span={12} mdSpan={8}>
```
Comment (christianvuerings) [inline docs/src/components/App.js:30]: Let's no longer use columns - something like `min-width` instead.

File: docs/src/components/Collapsable.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    components.includes(pathname.split('/')[2])
+  );
+
+  useEffect(() => {
+    setCollapsed(components.includes(pathname.split('/')[2]));
+  }, [components, pathname]);
+
+  return (
+    <>
+      {group === 'none' ? (
```
Comment (christianvuerings) [inline docs/src/components/Collapsable.js:27]: @AlbertCarreras let's use `group` === `null`

File: docs/src/components/Navigation.js [hunk 6]
```diff
@@ -1,57 +1,46 @@
...
 
 export default function Navigation() {
+  const gettingStarted = {};
+  const groups = {};
+  const { isSidebarOpen } = useSidebarContext();
+
+  Object.keys(routes).forEach(pathname => {
+    const { group, section } = routes[pathname].navRoute;
+
+    if (section === 'getting-started') {
```
Comment (christianvuerings) [inline docs/src/components/Navigation.js:16]: Cleanup

File: docs/src/components/Collapsable.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            onTap={() => {
+              setCollapsed(!collapsed);
+            }}
+          >
+            <Text size="lg" weight="bold">
+              {group}
+            </Text>
+          </TapArea>
+          {collapsed && (
+            <Box paddingY={2} marginStart={4} role="list">
```
Comment (christianvuerings) [inline docs/src/components/Collapsable.js:47]: padding / margin change

File: docs/src/components/Header.js [hunk 8]
```diff
@@ -1,11 +1,24 @@
...
+  Sticky,
+} from 'gestalt';
 import DocSearch from './DocSearch.js';
 import Link from './Link.js';
+import { useSidebarContext } from './SidebarContext.js';
 
 export default function Header() {
   const [isRTL, setIsRTL] = React.useState(false);
+  const { isSidebarOpen, setIsSidebarOpen } = useSidebarContext();
+  const handleSidebar = () => {
```
Comment (christianvuerings) [inline docs/src/components/Header.js:19]: @AlbertCarreras let's rename or inline

File: docs/src/components/NavLink.js [hunk 9]
```diff
@@ -17,21 +18,27 @@ const isModifiedEvent = event =>
...
     <Route path={to} location={history.location}>
-      {({ match }) => (
-        <Text weight={match ? 'bold' : 'normal'}>
-          <Link href={href} onClick={handleClick}>
-            {children}
-          </Link>
-        </Text>
-      )}
+      {({ match }) => {
+        return (
```
Comment (christianvuerings) [inline docs/src/components/NavLink.js:34]: @AlbertCarreras let's change this back

File: docs/src/components/Section.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+import { Box, TapArea, Text } from 'gestalt';
+import { useLocation } from 'react-router-dom';
+import Collapsable from './Collapsable.js';
+
+export default function Section({
+  section,
+  groups,
+}: {|
+  section: string,
+  groups: { [string]: Array<string> },
```
Comment (christianvuerings) [inline docs/src/components/Section.js:12]: @AlbertCarreras please use exact object syntax in here - `groups: {| [string]: Array<string> |}`

File: docs/src/components/Section.js [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+|}) {
+  const { pathname } = useLocation();
+  const [collapsed, setCollapsed] = useState(
+    section === pathname.split('/')[1]
+  );
+
+  useEffect(() => {
+    setCollapsed(section === pathname.split('/')[1]);
```
Comment (christianvuerings) [inline docs/src/components/Section.js:20]: @AlbertCarreras could you add a comment to what this does?

File: docs/src/components/Section.js [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <>
+      <TapArea
+        onTap={() => {
+          setCollapsed(!collapsed);
+        }}
+      >
+        <Box paddingY={2}>
+          <Text size="lg" weight="bold">
+            {section.replace(/-/g, ' ').toUpperCase()}
```
Comment (christianvuerings) [inline docs/src/components/Section.js:32]: @AlbertCarreras in the index file `getting-started` / `Getting Started`

File: docs/src/components/Section.js [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+          </Text>
+        </Box>
+      </TapArea>
+      {collapsed &&
+        Object.keys(groups).map(group => (
```
Comment (christianvuerings) [inline docs/src/components/Section.js:37]: @AlbertCarreras we can use `Object.entries` instead.
```
Object.entries(groups).map(([key, value]) => ...
```

File: docs/src/index.js [hunk 14]
```diff
@@ -17,13 +17,21 @@ if (container instanceof Element) {
...
-                render={() => <CardPage cards={routes[pathname]} />}
-              />
-            ))}
+            <Route
+              exact
+              path="/"
+              render={() => <Redirect to="/getting-started/Installation" />}
+            />
+            {Object.keys(routes).map(pathname => {
+              const cleanPathname = pathname.replace(/\d-/g, '');
```
Comment (christianvuerings) [inline docs/src/index.js:26]: @AlbertCarreras let's remove this

File: docs/src/components/App.js (hunk 4 above)
Comment (AlbertCarreras) [reply docs/src/components/App.js:30]: DONE

File: docs/src/components/Header.js (hunk 8 above)
Comment (AlbertCarreras) [reply docs/src/components/Header.js:19]: done

File: docs/src/components/NavLink.js (hunk 9 above)
Comment (AlbertCarreras) [reply docs/src/components/NavLink.js:34]: DONE

File: docs/src/components/Section.js (hunk 11 above)
Comment (AlbertCarreras) [reply docs/src/components/Section.js:20]: DONE

File: docs/src/components/Collapsable.js (hunk 7 above)
Comment (AlbertCarreras) [reply docs/src/components/Collapsable.js]: DONE

File: docs/src/1-Installation.doc.js (hunk 1 above)
Comment (AlbertCarreras) [reply docs/src/1-Installation.doc.js]: done

File: docs/src/1-Installation.doc.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/src/1-Installation.doc.js]: done

File: docs/src/components/Collapsable.js (hunk 5 above)
Comment (AlbertCarreras) [reply docs/src/components/Collapsable.js]: done

File: docs/src/components/Navigation.js (hunk 6 above)
Comment (AlbertCarreras) [reply docs/src/components/Navigation.js]: done

File: docs/src/components/Section.js (hunk 10 above)
Comment (AlbertCarreras) [reply docs/src/components/Section.js]: done

File: docs/src/components/Section.js (hunk 13 above)
Comment (AlbertCarreras) [reply docs/src/components/Section.js]: done

**Comment by christianvuerings** [comment]
@AlbertCarreras heads up - I did make a couple of changes:

## Fix focus outlines

Before
Screen Shot 2020-06-30 at 11 19 09 AM

After
Screen Shot 2020-06-30 at 11 19 18 AM

## Re-group components
Removed `Utilities` and moved `Link` to `Navigation`

## Fix search
Search no longer worked because we were loading search twice (because of the responsive layout)

## Padding
Several spacing fixes to line the sidebar up

## Accessibility fix
Instead of `Gestalt Documentation Menu` - we now use `Show/Hide Menu`

**Pull request merged**